cacerts/build.go

```diff
@@ -0,0 +1,68 @@
+package cacerts
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type Build struct {
+	Logger bard.Logger
+}
+
+// Build returns a libcnb.BuildResult for the given context. Build always contributes a launch layer containing the
+// ca-cert-helper executable.
+//
+// If the buildpack plan contains an entry with name "ca-certificates" Build will contribute a build layer
+// that adds the ca certificates at the paths provided in the plan entry metadata to the system truststore.
+func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
+	var result libcnb.BuildResult
+
+	b.Logger.Title(context.Buildpack)
+
+	for _, e := range context.Plan.Entries {
+		if e.Name != "ca-certificates" {
+			return libcnb.BuildResult{}, fmt.Errorf("received unexpected build plan entry %q", e.Name)
+		}
+		certPaths, err := pathFromEntryMetadata(e.Metadata)
+		if err != nil {
+			return libcnb.BuildResult{}, fmt.Errorf("failed to decode CA certificate paths from plan entry:\n%w", err)
+		}
+		layer := NewLayer(certPaths)
```

Comment:
I really don't like the idea of multiple layers for this, especially when its not necessary.  I think you should be iterating over the entries to build the complete list of paths, and then creating a *single* layer with those in it.

```go
func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
	b.Logger.Title(context.Buildpack)
	var result libcnb.BuildResult

	var paths []string

	for _, e := range context.Plan.Entries {
		if e.Name != "ca-certificates" {
			continue
		}

		p, err := extractPaths(e)
		if err != nil {
			return libcnb.BuildResult{}, fmt.Errorf("unable to extract paths\n%w", err)
		}

		paths = append(paths, p...)
	}

	if len(paths) > 0 {
		layer := NewLayer(paths)
		layer.Logger = b.Logger
		result.Layers = append(result.Layers, layer)
	}

	h := libpak.NewHelperLayerContributor(
		context.Buildpack,
		&context.Plan,
		"ca-cert-helper",
	)
	h.Logger = b.Logger
	result.Layers = append(result.Layers, h)

	return result, nil
}

func extractPaths(entry libcnb.BuildpackPlanEntry) ([]string, error) {
	raw, ok := entry.Metadata["paths"]
	if !ok {
		return nil, nil
	}
	p, ok := raw.([]string)
	if !ok {
		return nil, fmt.Errorf("inable to convert %+v to []string", raw)
	}

	return p, nil
}
```

Reply:
We aren't likely to fall into a case where there are multiple layers because the detector merges multiple bindings into a single plan entry but I agree in general.